File: Count the number of possible triangles.cpp
class Solution {
  public:
    // Function to count the number of possible triangles.
    int countTriangles(vector<int>& arr) {
        
        sort(arr.begin(),arr.end());
        int ans =0;
        int n = arr.size();
        for(int i = 0 ; i < n-2 ;i++ ){
            
            for(int j = i+1 ; j < n-1 ; j++){
                
                for(int k = j+1; k < n ; k++){
                    
                    if(arr[i]+arr[j] > arr[k]){
                        ans++;
                    } else break;
                }
            }
        }
        return ans;
    }
};

File: Count the number of possible triangles.java
class Solution {
    public int countTriangles(int arr[]) {
          int n = arr.length;
        Arrays.sort(arr);
        int count = 0;

        for (int i = 0; i < n - 2; i++) {
            int k = i + 1;
            for (int j = i + 2; j < n; j++) {
                while (k < j && arr[i] + arr[k] <= arr[j]) {
                    k++;
                }
                if (k < j) {
                    count += (j - k);
                }
            }
        }
        return count;
    }
}
